New package: Friction.Friction version 0.9.4

Dragon1573 opened the pull request:
## What's changed

- Resolve #119024
- Related https://github.com/friction2d/friction/issues/76

## Checks

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/122156)

Dragon1573 committed:
New package: Friction.Friction version 0.9.4

Dragon1573 committed:
Fixup: property "InstallerUrl"

Dragon1573 committed:
Fixup: InstallerUrl

Dragon1573 commented:
## Claim

This package CAN be installed silently and unattededly. Please re-run the Azure Pipelines and verify if there are difference between pipelines and locally verification.

## Details

```text
PS (local)> .\Tools\SandboxTest.ps1 .\manifests\f\Friction\Friction\0.9.4\
--> Validating Manifest
清单验证成功。

--> Checking dependencies
--> Starting Windows Sandbox, and:
    - Mounting the following directories:
      - C:\Users\Dragon1573\AppData\Local\Temp\SandboxTest as read-only
      - D:\Repository\winget-pkgs as read-and-write
    - Installing WinGet
    - Configuring Winget
    - Installing the Manifest 0.9.4
    - Refreshing environment variables
    - Comparing ARP Entries
```

```text
--> Installing WinGet
--> Disabling safety warning when running installer
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Enabled admin setting 'LocalManifestFiles'.
Enabled admin setting 'LocalArchiveMalwareScanOverride'.


--> Installing the Manifest 0.9.4

Found Friction [Friction.Friction] Version 0.9.4
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/friction2d/friction/releases/download/v0.9.4/friction-0.9.4-setup-win64.exe
  ██████████████████████████████  25.0 MB / 25.0 MB
Successfully verified installer hash
Starting package install...
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName            DisplayVersion Publisher ProductCode                                Scope
-----------            -------------- --------- -----------                                -----
Friction version 0.9.4 0.9.4          Friction  {12D25577-88FD-4C5A-AF69-CCA597CAC44E}_is1 User
```

![image](https://github.com/microsoft/winget-pkgs/assets/49941141/d97a7f73-2962-4bb5-b27b-60e8f3b020ba)

Trenly commented on manifests/f/Friction/Friction/0.9.4/Friction.Friction.installer.yaml:
```diff
@@ -0,0 +1,31 @@
+  InstallerUrl: https://github.com/friction2d/friction/releases/download/v0.9.4/friction-0.9.4-setup-win64.exe
+  InstallerSha256: 63D84A44B2BE5A9921720D782FCD5C7294863FB4BBD54BADE441E0A18C0055B9
+  InstallerSwitches:
+    Custom: /ALLUSER
```
```suggestion
    Custom: /ALLUSERS
```

Trenly commented on manifests/f/Friction/Friction/0.9.4/Friction.Friction.installer.yaml:
```diff
@@ -0,0 +1,31 @@
+  AppsAndFeaturesEntries:
+  - DisplayName: Friction version 0.9.4
+    Publisher: Friction
+    DisplayVersion: 0.9.4
```
```suggestion
```
DisplayVersion should be left off if it's the same as PackageVersion

Trenly commented on manifests/f/Friction/Friction/0.9.4/Friction.Friction.installer.yaml:
```diff
@@ -0,0 +1,31 @@
+  AppsAndFeaturesEntries:
+  - DisplayName: Friction version 0.9.4
+    Publisher: Friction
+    DisplayVersion: 0.9.4
```
```suggestion
```

Dragon1573 committed:
Fixup: Correct the switches

- InstallerSwitches > Custom: Should be "/ALLUSERS" for machine scope.
- Add the "S" suffix.

Dragon1573 committed:
Fixup: Remove "DisplayVersion"

- DisplayVersion should be left off if it's the same as PackageVersion.

stephengillie commented:
Manual Validation ended with: 
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/741babd9-a9bc-4d15-b674-da4fa7eae0de)

stephengillie commented on manifests/f/Friction/Friction/0.9.4/Friction.Friction.installer.yaml:
```diff
@@ -0,0 +1,29 @@
+- Architecture: x64
+  Scope: user
+  InstallerUrl: https://github.com/friction2d/friction/releases/download/v0.9.4/friction-0.9.4-setup-win64.exe
+  InstallerSha256: 63D84A44B2BE5A9921720D782FCD5C7294863FB4BBD54BADE441E0A18C0055B9
```
```suggestion
  InstallerSha256: 63D84A44B2BE5A9921720D782FCD5C7294863FB4BBD54BADE441E0A18C0055B9
  Dependencies:
    PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2015+.x64
```

stephengillie commented on manifests/f/Friction/Friction/0.9.4/Friction.Friction.installer.yaml:
```diff
@@ -0,0 +1,29 @@
+
+PackageIdentifier: Friction.Friction
+PackageVersion: 0.9.4
+InstallerType: inno
```
```suggestion
InstallerType: inno
Dependencies:
  PackageDependencies:
  - PackageIdentifier: Microsoft.VCRedist.2015+.x64
```

stephengillie commented:
With the dependency added, it encounters a pipeline limitation:
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/df310dd4-af9c-4f3f-a634-62a67aa08552)

- https://github.com/microsoft/winget-pkgs/issues/107639

Dragon1573 commented:
Sorry, I can't get it...

 Does it means this package requires a new feature of WinGet?  The feature isn't available now, so it blocks the Azure Pilepines?

Dragon1573 committed:
Fixup: Dependencies

- This package is dependent on "MSVCP140.dll" file, which provided by "Microsoft.VSRedist.2015+.x64"

Co-authored-by: Stephen Gillie <114952565+stephengillie@users.noreply.github.com>

stephengillie commented:
@wingetbot waivers Add Validation-Executable-Error

stephengillie commented:
> Sorry, I can't get it...
> 
> Does it means this package requires a new feature of WinGet? The feature isn't available now, so it blocks the Azure Pilepines?

Sorry for being unclear. This is a limitation of the current pipelines, not with your package.
